File: src/main/java/github/zmilla93/core/data/StashTabData.java
package github.zmilla93.core.data;

import github.zmilla93.core.enums.MatchType;
import github.zmilla93.gui.options.stash.StashTabType;

public class StashTabData {

    public final String stashTabName;
    public final MatchType matchType;
    public final StashTabType stashTabType;
    public final int stashColorIndex;

    public StashTabData(String name, MatchType matchType, StashTabType stashTabType, int stashColorIndex) {
        this.stashTabName = name;
        this.matchType = matchType;
        this.stashTabType = stashTabType;
        this.stashColorIndex = stashColorIndex;
    }

}

File: src/main/java/github/zmilla93/core/data/StashTabData.kt
package github.zmilla93.core.data

import github.zmilla93.core.enums.MatchType
import github.zmilla93.gui.options.stash.StashTabType

class StashTabData {
    @JvmField
    val stashTabName: String

    @JvmField
    val matchType: MatchType

    @JvmField
    val stashTabType: StashTabType

    @JvmField
    val stashColorIndex: Int

    /** GSON Constructor  */
    constructor() : this("", MatchType.EXACT_MATCH, StashTabType.NORMAL, 0)

    constructor(name: String, matchType: MatchType, stashTabType: StashTabType, stashColorIndex: Int) {
        this.stashTabName = name
        this.matchType = matchType
        this.stashTabType = stashTabType
        this.stashColorIndex = stashColorIndex
    }
}
